Pat_SO3.cpp

```
@@ -0,0 +1,66 @@
+#include "Pat_SO3.h"
+#include <iostream>
+#include <cmath>
+#include "Eigen/Dense"
+
+Pat_SO3::Pat_SO3(Eigen::Matrix3f R) {
+    if (is_SO3(R)) {
+        _rot = R;
+    } else {
+        _rot = Eigen::Matrix3f::Identity();
```

I don't think I would fall back to default constructing a matrix here - that will make it really hard to debug when you pass an invalid matrix. Better to just throw an exception.